lib/puppet/provider/kernel_parameter/grub2.rb

```diff
@@ -34,7 +34,7 @@ def self.mkconfig_cmdline
     # Fedora and Amazon Linux lack support and are excluded
     # since they don't have a release with major version 9
     needs_bls_cmdline = os.is_a?(Hash) && os['family'] == 'RedHat' &&
-                        os['release']['major'].to_i == 9 && os['release']['minor'].to_i >= 3
+                        ((os['release']['major'].to_i == 9 && os['release']['minor'].to_i >= 3) || os['release']['major'].to_i > 10)
```

Comment:
Should be greater than or equal to 10, right? Greater than 10 would only match EL11 and newer.

```suggestion
                        (os['release']['major'].to_i == 9 && os['release']['minor'].to_i >= 3) || os['release']['major'].to_i >= 10
```